tuf/data/targets.go

```diff
@@ -43,7 +42,7 @@ func isValidTargetsStructure(t Targets, roleName RoleName) error {
 	}
 
 	for _, roleObj := range t.Delegations.Roles {
-		if !IsDelegation(roleObj.Name) || path.Dir(roleObj.Name.String()) != roleName.String() {
+		if !IsDelegation(roleObj.Name) {
```

Comment:
I was trying to get away from the strict hierarchy, looking forward to multi-role delegation. The DFS for delegations just needs to check for a cycle and otherwise doesn't really need to care about the names. Is there a reason for enforcing the hierarchy that I'm missing?